crates/sui/src/sui_commands.rs

```diff
@@ -42,17 +49,101 @@ use sui_swarm_config::network_config_builder::ConfigBuilder;
+    ///
+    /// Note that if you want to start an indexer, Postgres DB is required.
     #[clap(name = "start")]
     Start {
+        /// Config directory that will be used to store network config, node db, keystore
+        /// sui genesis -f --with-faucet generates a genesis config that can be used to start this
+        /// proces. Use with caution as the `-f` flag will overwrite the existing config directory.
+        /// We can use any config dir that is generated by the sui genesis.
         #[clap(long = "network.config")]
-        config: Option<PathBuf>,
+        config_dir: Option<std::path::PathBuf>,
+
+        /// A new genesis is created each time this flag is set, and state is not persisted between
+        /// runs. Only use this flag when you want to start the network from scratch every time you
+        /// run this command.
+        #[clap(long)]
+        force_regenesis: bool,
+
+        /// Start a GraphQL server with default host and port: 127.0.0.1:8000. Note that GraphQL
+        /// requires a running indexer.
+        #[clap(long)]
+        with_graphql: bool,
+
+        /// Start a faucet with default host and port: 127.0.0.1:9123.
+        #[clap(long)]
+        with_faucet: bool,
+
+        /// Start an indexer with default host and port: 0.0.0.0:9124.
+        /// The indexer will be started in writer mode and reader mode.
+        #[clap(long)]
+        with_indexer: bool,
+
+        /// Port to start the Fullnode RPC server on. Default port is 9000.
+        #[clap(long, default_value = "9000")]
+        fullnode_rpc_port: u16,
+
+        /// Port to start the faucet on. Default port is 9123.
+        #[clap(long, default_value = "9123")]
+        faucet_port: u16,
+
+        /// Host to start the GraphQl server on. Default host is 127.0.0.1.
+        #[clap(long, default_value = "127.0.0.1")]
+        graphql_host: String,
+
+        /// Port to start the GraphQl server on. Default port is 8000.
+        #[clap(long, default_value = "8000")]
+        graphql_port: u16,
+
+        /// Port to start the Indexer RPC server on. Default port is 9124.
+        #[clap(long, default_value = "9124")]
+        indexer_rpc_port: u16,
+
+        /// Port for the Indexer Postgres DB. Default port is 5432.
+        #[clap(long, default_value = "5432")]
+        pg_port: u16,
+
+        /// Hostname for the Indexer Postgres DB. Default host is localhost.
+        #[clap(long, default_value = "localhost")]
+        pg_host: String,
+
+        /// DB name for the Indexer Postgres DB. Default DB name is sui_indexer.
+        #[clap(long, default_value = "sui_indexer")]
+        pg_db_name: String,
+
+        /// DB username for the Indexer Postgres DB. Default username is postgres.
+        #[clap(long, default_value = "postgres")]
+        pg_user: String,
+
+        /// DB password for the Indexer Postgres DB. Default password is postgrespw.
+        #[clap(long, default_value = "postgrespw")]
+        pg_password: String,
```

Comment:
It feels like this should be an args struct that the indexer already exposes, which we should be able inline into this args struct.

Reply:
The indexer config has a lot more fields than this, so not sure we want to do that?

Reply:
We can factor out the common args and inline them into both places.

Reply:
(But this can be a follow-up).